app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  attr_reader :storage_root, :druids
+  attr_reader :storage_root
 
   # @params [Hash] params used to initialize the Reporter service
   # @return [Reporter] the reporter
   def initialize(params)
     @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
-    @druids = moab_storage_root_druid_list
+    @msr_names = {}
   end
 
-  # @return [Array] an array of druids on the storage root
-  def moab_storage_root_druid_list
-    druid_array = []
+  def moab_storage_root_list_pres_obj_comp_moabs_relation
     PreservedObject
       .joins(:complete_moabs)
       .where(complete_moabs: { moab_storage_root: storage_root })
-      .select(:druid)
       .order(:druid)
+  end
+
+  # @return [Array] an array of druids on the storage root
+  def druid_csv_list
+    druid_array = [['druid']]
+    moab_storage_root_list_pres_obj_comp_moabs_relation
+      .select(:druid)
       .each_row do |po_hash|
```

Comment:
I'm not that familiar with the `#each_row` implementation. Will that loop over rows in the DB without loading it all into memory? Maybe this is also what @justinlittman was asking.

Reply:
`#each_row` is from this gem:  https://github.com/afair/postgresql_cursor

it allows the result set to be returned in pages using a real database cursor, without the jankiness of `find_each`.  in the #1347 implementation, it _was_ used to address the concern of loading everything into memory before writing, but that was lost in the DRYing of #1360.  i could go either way on whether we should spend effort on getting back to writing the CSV out within the query paging instead of loading the full result set into memory.

Reply:
I am totally fine with going forward with this and optimizing for performance/robustness later on if it is determined that this is *actually* problematic.